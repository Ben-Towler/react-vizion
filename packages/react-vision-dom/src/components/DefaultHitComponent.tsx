import { SearchResult } from './HitsComponent';

type Props = {
  searchResult: SearchResult;
};

const DefaultHitComponent: React.FunctionComponent<Props> = ({
  searchResult,
}) => {
  return (
    <div className="default-hit-card">
      <div className="default-hit-container">
        <h4 className="default-hit-title">{searchResult.resourceName}</h4>
        <div className="card-body">{JSON.stringify(searchResult, null, 2)}</div>
      </div>
    </div>
  );
};

export default DefaultHitComponent;
