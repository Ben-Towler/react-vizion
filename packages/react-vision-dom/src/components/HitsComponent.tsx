import React from 'react';
import DefaultHitComponent from './DefaultHitComponent';

export type SearchResult = {
  resourceId: string;
  address: any;
  distance: number;
  geoPoint: {
    lat: number;
    lng: number;
  };
  resourceName: string;
  note: string;
  onlineBookingUrl: string;
  phones: any;
  type: string;
};

type Props = {
  results: SearchResult[];
  className?: string;
  style?: React.CSSProperties;
  noResultsFound?: React.ReactNode;
  hit?: (result: SearchResult) => React.ReactNode;
};

const HitsComponent: React.FunctionComponent<Props> = ({
  results,
  className,
  style,
  hit,
  noResultsFound,
}) => {
  return (
    <div className={className} style={style}>
      {Array.isArray(results) && results.length > 0 ? (
        results.map(searchResult => (
          <div key={searchResult.resourceId}>
            {hit ? (
              hit(searchResult)
            ) : (
              <>
                <DefaultHitComponent searchResult={searchResult} />
              </>
            )}
          </div>
        ))
      ) : (
        <>
          {noResultsFound || (
            <div className="default-hit-no-results-found"></div>
          )}
        </>
      )}
      ;
    </div>
  );
};

export default HitsComponent;
