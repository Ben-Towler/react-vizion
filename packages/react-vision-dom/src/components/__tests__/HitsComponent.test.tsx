import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HitsComponent from '../HitsComponent';
import DefaultHitComponent from '../DefaultHitComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('HitsComponent', () => {
  const results = [{ resourceId: '123', resourceName: 'test' }];

  it('applies its default props', () => {
    //@ts-ignore
    const instance = renderer.create(<HitsComponent refine={() => null} />);

    expect(instance.toJSON()).toMatchSnapshot();

    instance.unmount();
  });

  it('should render with default hit component', () => {
    const wrapper = shallow(<HitsComponent results={results} />);
    const defaultHitComponent = (
      <DefaultHitComponent searchResult={results[0]} />
    );

    expect(wrapper.containsMatchingElement(defaultHitComponent)).toBeTruthy();
    expect(wrapper.exists('.default-hit-no-results-found')).toBeFalsy();
  });

  it('should render with default no results found', () => {
    const wrapper = shallow(<HitsComponent results={[]} />);

    expect(wrapper.exists('.default-hit-no-results-found')).toBeTruthy();
  });

  describe('Style tests', () => {
    it('should render main div with className props ', () => {
      const classNameTest = 'class-test';
      const wrapper = shallow(
        <HitsComponent className={classNameTest} results={[]} />
      );

      expect(wrapper.first('div').hasClass(classNameTest)).toBeTruthy();
    });

    it('should render main div with style props ', () => {
      const styleTest = { top: 0 };
      const wrapper = shallow(<HitsComponent style={styleTest} results={[]} />);

      expect(wrapper.first('div').props().style).toEqual(styleTest);
    });
  });

  describe('Custom inner components', () => {
    it('should render with custom hit component', () => {
      const customHitComponent = <h1>Test</h1>;
      const wrapper = shallow(
        <HitsComponent results={results} hit={_ => customHitComponent} />
      );

      expect(wrapper.contains(customHitComponent)).toBeTruthy();
      expect(wrapper.exists('.default-hit-no-results-found')).toBeFalsy();
    });

    it('should render with custom noResultsFound component', () => {
      const customNoResultsFound = <h1>Test</h1>;

      const wrapper = shallow(
        <HitsComponent noResultsFound={customNoResultsFound} results={[]} />
      );

      expect(wrapper.contains(customNoResultsFound)).toBeTruthy();
    });
  });
});
