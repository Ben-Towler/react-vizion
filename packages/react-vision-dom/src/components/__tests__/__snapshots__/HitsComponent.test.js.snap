// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HitsComponent applies its default props 1`] = `
<div>
  <div
    className="default-hit-no-results-found"
  />
  ;
</div>
`;
